packages/edit-site/src/components/left-sidebar/navigation-panel/template-switcher/index.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+import {
+	Tooltip,
+	__experimentalNavigationGroup as NavigationGroup,
+	__experimentalNavigationItem as NavigationItem,
+} from '@wordpress/components';
+import { Icon, home, plus, undo } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import TemplatePreview from './template-preview';
+import ThemePreview from './theme-preview';
+
+const TEMPLATE_OVERRIDES = {
+	page: ( slug ) => `page-${ slug }`,
+	category: ( slug ) => `category-${ slug }`,
+	post: ( slug ) => `single-post-${ slug }`,
+};
+
+function TemplateLabel( { template, homeId } ) {
+	return (
+		<>
+			{ template.slug }{ ' ' }
+			{ template.id === homeId && (
+				<Tooltip text={ __( 'Home' ) }>
+					<div className="edit-site-template-switcher__label-home-icon">
+						<Icon icon={ home } />
+					</div>
+				</Tooltip>
+			) }
+			{ template.status !== 'auto-draft' && (
+				<Tooltip text={ __( 'Customized' ) }>
+					<span className="edit-site-template-switcher__label-customized-dot" />
+				</Tooltip>
+			) }
+		</>
+	);
+}
```

Comment:
This looks very similar to `NavigationItemWithIcon`, could we "merge" the 2 components?

Reply:
Kinda different 🤔 I unified them but I'm not sure if it's any better this way. Both way seems a little bit confusing. Maybe it's too smart now.